test/common/network/apple_dns_impl_test.cc

```diff
@@ -0,0 +1,208 @@
+#include <list>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/common/platform.h"
+#include "envoy/config/core/v3/address.pb.h"
+#include "envoy/event/dispatcher.h"
+#include "envoy/network/address.h"
+#include "envoy/network/dns.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/utility.h"
+#include "common/event/dispatcher_impl.h"
+#include "common/network/address_impl.h"
+#include "common/network/apple_dns_impl.h"
+#include "common/network/filter_impl.h"
+#include "common/network/listen_socket_impl.h"
+#include "common/network/utility.h"
+#include "common/runtime/runtime_impl.h"
+#include "common/stream_info/stream_info_impl.h"
+
+#include "test/mocks/common.h"
+#include "test/mocks/local_info/mocks.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/protobuf/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/thread_local/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/printers.h"
+#include "test/test_common/utility.h"
+
+#include "absl/container/fixed_array.h"
+#include "absl/container/node_hash_map.h"
+#include "gtest/gtest.h"
+
+using testing::_;
+using testing::Contains;
+using testing::InSequence;
+using testing::IsSupersetOf;
+using testing::NiceMock;
+using testing::Not;
+using testing::Return;
+
+namespace Envoy {
+namespace Network {
+namespace {
+
+// Note: this test suite is, unfortunately, not hermetic. Apple's APIs do not allow overriding the
```

Comment:
I thought we discussed not doing the internet based testing? This seems doomed to flake and only works because it's an OSX test. I thought you had some other way to hack/avoid this or just do localhost based testing?

Reply:
No, that was the problem -- no way that I could find to point to a local resolver :/

Reply:
No I mean just run tests against "localhost" or whatever using the default resolver. Relying on internet resolution seems not good, but I guess we can see how often it flakes in CI.

Reply:
Ah I see, yeah I resolve localhost in another test. I'll keep this, and be vigilant about flakes.